apps/betterangels-backend/clients/schema.py

```diff
@@ -341,9 +342,29 @@ def client_profile(self, info: Info, pk: strawberry.ID) -> ClientProfileType:
         extensions=[HasRetvalPerm(AttachmentPermissions.VIEW)],
     )
 
-    client_documents: OffsetPaginated[ClientDocumentType] = strawberry_django.offset_paginated(
-        extensions=[HasRetvalPerm(AttachmentPermissions.VIEW)],
-    )
+    @strawberry_django.offset_paginated(extensions=[HasPerm(AttachmentPermissions.VIEW)])
+    def client_documents(
+        self, info: Info, client_id: str, document_groups: Optional[list[ClientDocumentGroupEnum]] = None
+    ) -> OffsetPaginated[ClientDocumentType]:
+        current_user = cast(User, get_current_user(info))
+
+        content_type = ContentType.objects.get_for_model(ClientProfile)
+
+        namespaces = []
+        if document_groups:
+            namespaces = [ns.value for group in document_groups for ns in CLIENT_DOCUMENT_NAMESPACE_GROUPS[group]]
+
+        documents: OffsetPaginated[ClientDocumentType] = filter_for_user(
```

Comment:
Is there a way to layer on the existing filtering logic so we don't have to update the function each time we add a filter param?

Reply:
moved to filter